File: backend/models/Driver.js
const mongoose = require('mongoose');

const DriverSchema = new mongoose.Schema({
    name: { type: String, required: true },
    phone: { type: String, required: true }, // Unique per organization, not globally
    password: { type: String, required: true, default: '$2a$10$YourHashedPasswordHere' }, // Will be properly hashed
    vehicleModel: { type: String }, // Optional
    vehicleNumber: { type: String }, // Optional
    vehicleCategory: {
        type: String,
        // Enum validation removed or extended to allow Logistics types if strictness passed
    },
    organizationId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Organization',
        required: true,
        index: true
    },
    vertical: {
        type: String,
        enum: ['TAXI', 'LOGISTICS'],
        default: 'TAXI',
        required: true
    },
    status: {
        type: String,
        enum: ['ONLINE', 'OFFLINE', 'BUSY'],
        default: 'OFFLINE'
    },
    currentLocation: {
        lat: Number,
        lng: Number
    },
    rating: { type: Number, default: 5.0 },
    // Logistics Specific Fields
    lorryName: String,
    ownerName: String,
    ownerPhone: String,
    ownerHometown: String,
    panNumber: String,

    // Fleet Compliance Documents
    fcStatus: Date,       // Fitness Certificate Expiry
    insuranceExpiry: Date,
    taxExpiry: Date,

    isActive: { type: Boolean, default: true },

    bankDetails: {
        accountName: String,
        bankName: String,
        accountNumber: String,
        ifsc: String,
        upiNumber: String
    }
});

// Phone number unique per organization
DriverSchema.index({ organizationId: 1, phone: 1 }, { unique: true });

module.exports = mongoose.model('Driver', DriverSchema);

File: setgo_driver_backup/lib/models/driver.dart
class Driver {
  final String id;
  final String name;
  final String phone;
  final String vehicleModel;
  final String vehicleNumber;
  final String status;
  final String organizationId;
  final String? organizationName;
  final String? token;

  Driver({
    required this.id,
    required this.name,
    required this.phone,
    required this.vehicleModel,
    required this.vehicleNumber,
    required this.status,
    required this.organizationId,
    this.organizationName,
    this.token,
  });

  factory Driver.fromJson(Map<String, dynamic> json) {
    return Driver(
      id: json['_id'],
      name: json['name'],
      phone: json['phone'],
      vehicleModel: json['vehicleModel'],
      vehicleNumber: json['vehicleNumber'],
      status: json['status'],
      organizationId: json['organizationId'],
      organizationName: json['organizationName'],
      token: json['token'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'phone': phone,
      'vehicleModel': vehicleModel,
      'vehicleNumber': vehicleNumber,
      'status': status,
      'organizationId': organizationId,
      'organizationName': organizationName,
      'token': token,
    };
  }
}
